app/controllers/merchant/items_controller.rb

```diff
@@ -7,45 +9,95 @@ def index
     @merchant = Merchant.where("id = #{current_user.merchant.id}").first
   end
 
+  def edit
+    @item = if session[:failed_update]
+              Item.find(session[:failed_update])
+            else
+              Item.find(params[:id])
+            end
+  end
+
+  def update
+    @item = Item.find(params[:id])
+    if params[:item]
+      update_item(@item)
+    else
+      @item.switch_active_status
+      switch_active_with_flash(@item)
+    end
+  end
+
+  def destroy
+    item = Item.find(params[:id])
+    destroy_item(item)
+  end
+
+  def new
+    @item = if session[:failed_save]
+              Item.new(session[:failed_save])
+            else
+              Item.new
+            end
+  end
+
   def create
-    merchant = Merchant.where("id = #{current_user.merchant.id}").first
+    merchant = Merchant.find(current_user.merchant.id)
     @item = merchant.items.create(item_params)
```

Comment:
This seems like a cleaner way of grabbing the merchant

Reply:
Agreed!